spec/utils/ssl.rb

```diff
@@ -142,6 +142,59 @@ def with_files_in(tmpdir, &block)
       block.call(bundle_file, key_file, chain_file, config_file)
     end
 
+    def with_files_in_default_dirs(tmpdir, &block)
+      fixtures_dir = File.join(tmpdir, 'fixtures')
+      default_ssldir = File.join(ENV['HOME'], '.puppetlabs', 'etc', 'puppet', 'ssl')
+      default_cadir = File.join(ENV['HOME'], '.puppetlabs', 'etc', 'puppetserver', 'ca')
+
+      FileUtils.mkdir_p fixtures_dir
+      # Clean up anything already here
+      FileUtils.remove_dir(default_ssldir, true)
```

Comment:
Soo I'm actually not sure how I feel about this. This will clean up anything I have in my normal dev dir as a side effect of running these tests, which I feel could be unexpected and possibly destructive. But if others think we should treat that dev dir as unreliable (i.e. anything could mess with it at any time), I'll defer to that.

Reply:
We could potentially back up what's there and put it back at the end?

Reply:
@justinstoller @tvpartytonight Do y'all have opinions on this?

Reply:
I feel more ok about modifying the `puppetserver/ca` directory, and less ok about deleting the `puppet/ssl` directory. I think that it is quite reasonable for this repo to modify the contents of `puppetserver/ca`, but `puppet/ssl` seems more of an unexpected side effect.

Not sure if this applies here, but could the structure of this be changed to allow for the tests to modify different locations, while the actual production code makes that impossible? For instance, the actual [migrate](https://github.com/puppetlabs/puppetserver-ca-cli/blob/main/lib/puppetserver/ca/action/migrate.rb#L57) call in `run` only ever calls it with 1 argument, but the actual method [definition](https://github.com/puppetlabs/puppetserver-ca-cli/blob/main/lib/puppetserver/ca/action/migrate.rb#L66-L69) has a default that can be overridden, the sole purpose for [easier testing](https://github.com/puppetlabs/puppetserver-ca-cli/blob/main/spec/puppetserver/ca/action/migrate_spec.rb#L85-L92).

Reply:
hmm, I'm not sure if it applies either, since the way I'm testing this doesn't call `symlink_to_old_cadir` directly. But I'll think about this. And we can do Maggie's backup suggestion if I can't come up with something better.